source/store/actions/dynamicFormData.ts

```diff
@@ -0,0 +1,103 @@
+import { Form } from '../../types/FormTypes';
+import { Case } from '../../types/CaseType';
+import { User } from '../../types/UserTypes';
+import { convertAnswerArrayToObject } from '../../helpers/DataStructure';
+
+const addIds = (descriptor: string[], questionId: string, formId: string) =>
+  descriptor.map(s => {
+    if (s.split('.')[0] !== 'user' && questionId !== '') {
+      return `${formId}.${questionId}.${s}`;
+    } else if (s.split('.')[0] !== 'user' && questionId === '') {
+      return `${formId}.${s}`;
+    }
+    return s;
+  });
+
+/** Takes a form and generates the dataMap template used to fill the dynamical data.  */
+const makeDataMap = (form: Form) => {
+  const dataMap = {};
+  form.steps.forEach(step => {
+    if (!step.questions) {
+      return;
+    }
+    step.questions.forEach(qs => {
```

Comment:
Would avoid the use of acronym qs here and use question instead :)
For better readability